journal/journal.go

```diff
@@ -50,11 +50,11 @@ const (
 	PriDebug
 )
 
-var conn net.Conn
+var conn net.PacketConn
 
 func init() {
 	var err error
-	conn, err = net.Dial("unixgram", "/run/systemd/journal/socket")
+	conn, err = net.ListenPacket("unixgram", "")
```

Comment:
- About `ListenPacket` vs `ListenUnixgram`, I'm fine with both. `ListenPacket` was used to avoid the creation of a new `UnixAddr` required in the `ListenUnixgram` arguments.
- Concerning _DialFoo_, sorry, I'm not a golang developer. I don't see how it can be done wihout having a successful connection. The only way to do it was to bind and listen on a _random_ abns.
- About the `""` (empty) address, it will be addressed as "\0" on the system side, which will use the autobind feature of the kernel.